eng/common/TestResources/New-TestResources.ps1

```diff
@@ -827,6 +827,19 @@ try {
                                                                 -templateFile $templateFile `
                                                                 -environmentVariables $EnvironmentVariables
 
+        $storageAccounts = Retry { Get-AzResource -ResourceGroupName $ResourceGroupName -ResourceType "Microsoft.Storage/storageAccounts" }
+        if ($storageAccounts) {
+            $clientIp = Retry { Invoke-RestMethod -Uri 'https://myip.dnsomatic.com/' }  # OpenDNS owned ip site. Some throttling will occur if called quickly.
```

Comment:
This might work in most cases, but I don't think it's **guaranteed** to work, since your network might route your request over different public IPs depending on the destination.

Reply:
In that case I'm not sure we have a great workaround, other than to enable storage access logs, wait for them to appear, download the audit history, and parse the client IP out of it :)

Reply:
I'm also not happy taking a third party site dependency like this, as most sites I found seem to be hosted by unknown/random entities. I'd like to have a backup site to call if the main one doesn't work, but I couldn't find anything else I trusted as much as the opendns org. Would be nice if Microsoft had a site like this somewhere.

Reply:
Another option is we could run a simple app service to do this for ourselves with opendns as a backup.